tests/test_utils/test_local_file_mods.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+import pandas as pd
+import os
+import json
+import io
+from pathlib import Path
+
+from src.utils.local_file_mods import (
+    read_local_csv,
+    write_local_csv,
+    load_local_json,
+    local_file_exists,
+    local_file_size,
+    check_file_exists,
+    local_mkdir,
+    local_open,
+    local_file_write_feather,
+)
+
+
+@pytest.fixture
+def input_data():
+    data = {
+        "run_id": [1, 2],
+        "timestamp": ["Time:1", "Time:2"],
+        "version": ["0.0.0", "0.0.1"],
+        "duration": [5.0, 6.0],
+    }
+    return pd.DataFrame(data)
+
+
+@pytest.fixture
+def expout_data():
+    data = {
+        "run_id": [1, 2],
+        "timestamp": ["Time:1", "Time:2"],
+        "version": ["0.0.0", "0.0.1"],
+        "duration": [5.0, 6.0],
+    }
+    return pd.DataFrame(data)
+
+
+@pytest.fixture
+def test_csv_file(tmp_path, input_data):
+    # Write the input data to a csv for testing
+    filepath = tmp_path / "test.csv"
+    input_data.to_csv(filepath, index=False)
+    return filepath
+
+
+def test_read_local_csv(test_csv_file, expout_data):
+    # Creating df using the function and test csv
+    df = read_local_csv(str(test_csv_file))
+    # Make sure the reader function has returned a df
+    assert isinstance(df, pd.DataFrame)
+    # Check that the df is the same as the expected data
+    pd.testing.assert_frame_equal(df, expout_data)
+
+
+def test_write_local_csv(tmp_path, input_data):
+    filepath = tmp_path / "test.csv"
+
+    write_local_csv(str(filepath), input_data)
+
+    # Use os and pathlib to check that the written file exists
+    assert os.path.exists(filepath)
```

Comment:
The function write_local_csv doesn't check if the path exists. This is fine to keep but doesn't imitate what the function would end up doing

Reply:
https://jira.ons.gov.uk/browse/RDRP-265